src/models.ts

```diff
@@ -0,0 +1,112 @@
+const customSdkErrorEventCollection = [
+  "onEventTypeNotSupported",
+  "onEventTypeNotExpected",
+];
+export type CustomSdkErrorEvents = typeof customSdkErrorEventCollection[number];
+
+const generalEventCollection = [
```

Comment:
There's no need to append 'collection', 'array', 'list', 'string' to the values, it should be enough self explanatory since we are assigning array.

const generalEvents = [...]